### PR TITLE
Link attributes to a version of the document

### PR DESCRIPTION
**Pull request by akorneyc**
This PR include Link attributes to a version of the document.

### CODE REVIEW

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java [hunk 1]
```diff
@@ -109,4 +120,45 @@ public void setChecksumAlgo(String checksumAlgo) {
...
+    public List<DocumentAttributeEntity> getDocumentAttributes() {
+        return documentAttributes;
+    }
+
+    public void setDocumentAttributes(List<DocumentAttributeEntity> documentAttributes) {
+        this.documentAttributes = documentAttributes;
+    }
+
+    /**
+     * Associates an attribute, and it's value, to a document.
```
Comment (melissapalmer) [inline src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java]: Please update doc's, its not to the document anymore. But document version.

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java [hunk 2]
```diff
@@ -109,4 +120,45 @@ public void setChecksumAlgo(String checksumAlgo) {
...
+        DocumentAttributeEntity documentAttribute = new DocumentAttributeEntity(this, attribute, value);
+        documentAttribute.setUpdatedBy(addedBy);
+
+        documentAttributes.add(documentAttribute);
+
+        return documentAttribute;
+    }
+
+    /**
+     * Removes an attribute association from the document.
```
Comment (melissapalmer) [inline src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java]: Please update doc's, its not to the document anymore. But document version.

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionService.java [hunk 3]
```diff
@@ -37,4 +40,29 @@
 	 */
 	public DocumentVersionEntity createDocumentVersion(DocumentEntity document, String checksum, String checksumAlgo, String documentType);
 
+	
+    /**
+     * Adds an attribute association to a document with a value.
```
Comment (melissapalmer) [inline src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionService.java]: Please update doc's, its not to the document anymore. But document version.

**Comment by akorneyc** [comment]
> 
> 
> Couple questions around this PR:
> 
>     1. What has been done in relation to the following question?
>        [#60 (comment)](https://github.com/GrindrodBank/dokuti/issues/60#issuecomment-584493957)
> 
>     2. How does this change affect the filter option `filterAttributes` on GET `/documents`?
> 
>     3. Does this change have any impact on the UI I wonder?
> 
> 
> Thanks
1) for now new uploaded version has no attributes , need clear up requirements
2) filterAttributes must applyies to latest(and current) version. Actually current API allow to download latest(current) version
3) no idea about UI but it seems yes as API  allow to change attributes of any version


> 
> 
> Couple questions around this PR:
> 
>     1. What has been done in relation to the following question?
>        [#60 (comment)](https://github.com/GrindrodBank/dokuti/issues/60#issuecomment-584493957)
> 
>     2. How does this change affect the filter option `filterAttributes` on GET `/documents`?
> 
>     3. Does this change have any impact on the UI I wonder?
> 
> 
> Thanks

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java (hunk 1 above)
Comment (akorneyc) [reply src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java]: fixed

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java (hunk 2 above)
Comment (akorneyc) [reply src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionEntity.java]: fixed

File: src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionService.java (hunk 3 above)
Comment (akorneyc) [reply src/main/java/za/co/grindrodbank/dokuti/documentversion/DocumentVersionService.java]: fixed

**Pull request merged**